test/python/primitives/test_sampler.py

```diff
@@ -783,6 +784,19 @@ def test_circuit_with_unitary(self):
         sampler_result = sampler.run([circuit]).result()
         self.assertDictAlmostEqual(sampler_result.quasi_dists[0], {0: 1, 1: 0})
 
+    def test_circuit_key_controlflow(self):
+        """Test for a circuit with control flow."""
+        qc = QuantumCircuit(2, 1)
+
+        with qc.for_loop(range(5)):
+            qc.h(0)
+            qc.cx(0, 1)
+            qc.measure(0, 0)
+            qc.break_loop().c_if(0, True)
+
+        self.assertIsInstance(hash(_circuit_key(qc)), int)
+        self.assertIsInstance(json.dumps(_circuit_key(qc)), str)
+
```

Comment:
Aer supports control flow - would it be possible to try executing a simple `Sampler` and `Estimator` run on a `BackendX` backed by `AerSimulator` (with the appropriate `@unittest.skipUnless(optionals.HAS_AER)`).

Reply:
Thanks. I'll try it.